src/System.Private.CoreLib/shared/System/Resources/ManifestBasedResourceGroveler.cs

```diff
@@ -237,10 +237,23 @@ internal ResourceSet CreateResourceSet(Stream store, Assembly assembly)
                     }
                     else
                     {
-                        Type readerType = Type.GetType(readerTypeName, throwOnError: true);
-                        object[] args = new object[1];
-                        args[0] = store;
-                        IResourceReader reader = (IResourceReader)Activator.CreateInstance(readerType, args);
+                        IResourceReader reader;
+
+                        // Permit deserialization as long as the default ResourceReader is used
+                        if (ResourceManager.IsDefaultType(readerTypeName, ResourceManager.ResReaderTypeName))
+                        {
+                            reader = new ResourceReader(
```

Comment:
You could add an internal overload so that you don't have to pass in the dictionary, but this seems OK so long as others don't object to the duplication.